blocks/algorithms/__init__.py

```diff
@@ -575,6 +577,64 @@ def compute_steps(self, previous_steps):
         return steps, []
 
 
+class VariableClipping(StepRule):
+    """Clip the maximum norm of individual variables along certain axes.
+
+    This :class:`StepRule` can be used to implement L2 norm constraints on
+    individual hidden units, groups of hidden units, convolutional
```

Comment:
I am not sure I understand how it can applied to anything different from entire weight tensors.